grafana-loki,promtail: 3.4.3 -> 3.5.1

r-ryantm opened the pull request:
Automatic update generated by [nixpkgs-update](https://github.com/ryantm/nixpkgs-update) tools. This update was made based on information from passthru.updateScript.


meta.description for promtail is: Like Prometheus, but for logs

meta.homepage for promtail is: https://grafana.com/oss/loki/

meta.changelog for promtail is: https://github.com/grafana/loki/releases/tag/v3.5.0


###### Updates performed
- Ran passthru.UpdateScript

###### To inspect upstream changes

- [Release on GitHub](https://github.com/grafana/loki/releases/tag/v3.5.0)

- [Compare changes on GitHub](https://github.com/grafana/loki/compare/v3.4.3...v3.5.0)

###### Impact

<b>Checks done</b>

---

- built on NixOS

> [!CAUTION]
> A test defined in `passthru.tests` did not pass.
- found 3.5.0 with grep in /nix/store/pasibraayzwmmmwz1gm4f8zb07912a69-promtail-3.5.0
- found 3.5.0 in filename of file in /nix/store/pasibraayzwmmmwz1gm4f8zb07912a69-promtail-3.5.0

---

<details>
<summary>
<b>Rebuild report</b> (if merged into master) (click to expand)
</summary>

```
3 total rebuild path(s)

3 package rebuild(s)

First fifty rebuilds by attrpath

grafana-loki
promtail
```

</details>

<details>
<summary>
<b>Instructions to test this update</b> (click to expand)
</summary>

---

Either **download from the cache**:
```
nix-store -r /nix/store/pasibraayzwmmmwz1gm4f8zb07912a69-promtail-3.5.0 \
  --option binary-caches 'https://cache.nixos.org/ https://nixpkgs-update-cache.nix-community.org/' \
  --option trusted-public-keys '
  nixpkgs-update-cache.nix-community.org-1:U8d6wiQecHUPJFSqHN9GSSmNkmdiFW7GW7WNAnHW0SM=
  cache.nixos.org-1:6NCHdD59X431o0gWypbMrAURkbJ16ZPMQFGspcDShjY=
  '
```
(The nixpkgs-update cache is only trusted for this store-path realization.)
For the cached download to work, your user must be in the `trusted-users` list or you can use `sudo` since root is effectively trusted.

Or, **build yourself**:
```
nix-build -A promtail https://github.com/r-ryantm/nixpkgs/archive/b0aca4f221d9531bc1e32b03ae35cb6e06007019.tar.gz
```
Or:
```
nix build github:r-ryantm/nixpkgs/b0aca4f221d9531bc1e32b03ae35cb6e06007019#promtail
```

After you've downloaded or built it, look at the files and if there are any, run the binaries:
```
ls -la /nix/store/pasibraayzwmmmwz1gm4f8zb07912a69-promtail-3.5.0
ls -la /nix/store/pasibraayzwmmmwz1gm4f8zb07912a69-promtail-3.5.0/bin
```

---

</details>
<br/>



### Pre-merge build results

We have automatically built all packages that will get rebuilt due to
this change.

This gives evidence on whether the upgrade will break dependent packages.
Note sometimes packages show up as _failed to build_ independent of the
change, simply because they are already broken on the target branch.

## `nixpkgs-review` result

Generated using [`nixpkgs-review`](https://github.com/Mic92/nixpkgs-review).

Command: `nixpkgs-review`

---
### `x86_64-linux`
<details>
  <summary>:white_check_mark: 2 packages built:</summary>
  <ul>
    <li>grafana-loki</li>
    <li>promtail</li>
  </ul>
</details>

---

###### Maintainer pings

cc @WilliButz @globin @mmahut @emilylange for [testing](https://github.com/ryantm/nixpkgs-update/blob/main/doc/nixpkgs-maintainer-faq.md#r-ryantm-opened-a-pr-for-my-package-what-do-i-do).

> [!TIP]
> As a maintainer, if your package is located under `pkgs/by-name/*`, you can comment **`@NixOS/nixpkgs-merge-bot merge`** to automatically merge this update using the [`nixpkgs-merge-bot`](https://github.com/NixOS/nixpkgs-merge-bot).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

r-ryantm committed:
grafana-loki,promtail: 3.4.3 -> 3.5.0

https://github.com/grafana/loki/releases/tag/v3.5.0

diff: https://github.com/grafana/loki/compare/v3.4.3...v3.5.0

Co-authored-by: emilylange <git@emilylange.de>

adamcstephens commented:
This technically has breaking changes. One affecting kubernetes users, and another removing a deprecated configs. Are we waiting until after fork off to merge because of these? or should we proceed now?

emilylange commented:
No hard preference.
But note that the VM test is once again failing because Grafana Labs isn't keeping their example configuration in sync.

r-ryantm committed:
grafana-loki,promtail: 3.5.0 -> 3.5.1

https://github.com/grafana/loki/releases/tag/v3.5.1

diff: https://github.com/grafana/loki/compare/v3.5.0...v3.5.1

Co-authored-by: emilylange <git@emilylange.de>

emilylange commented:
> How do we proceed here, since I see you drafted the other PR too.

I marked both this PR and #411262 as draft because in both cases our VM test is failing and needs manual fixing.
I don't care who does it, but I probably won't find the time until later this week :)

adamcstephens reviewed (approved):
Thanks!
